cnudie/retrieve.py

```diff
@@ -37,6 +41,82 @@
     typing.Sequence[str]
 ]
 
+OcmRepositoryCfg = \
+    str | typing.Iterable[str] | \
+    cnudie.util.OcmLookupMappingConfig | \
+    cnudie.util.OcmLookupMapping | \
+    typing.Iterable[cnudie.util.OcmLookupMapping]
+
+
+OcmRepositoryLookup = typing.Callable[
+    [ComponentName],
+    typing.Generator[cm.OciRepositoryContext | str, None, None],
+]
+
+
+def _iter_ocm_repositories(
+    component: str | cm.ComponentIdentity | cm.Component,
+    repository_cfg: OcmRepositoryCfg,
+    /,
+):
+    if isinstance(component, cm.ComponentIdentity):
+        component = component.name
+    elif isinstance(component, cm.Component):
+        component = component.name
+
+    if isinstance(repository_cfg, cm.OciRepositoryContext):
+        yield repository_cfg
+        return
+
+    if isinstance(repository_cfg, str):
+        yield repository_cfg
+        return
+
+    if isinstance(repository_cfg, cnudie.util.OcmLookupMapping):
+        repository_cfg: cnudie.util.OcmLookupMapping
+        if repository_cfg.matches(component):
+            yield repository_cfg.ocm_repo_url
+        return
+
+    if isinstance(repository_cfg, cnudie.util.OcmLookupMappingConfig):
+        yield repository_cfg.iter_ocm_repositories(component)
+
+    # recurse into elements in case repository_cfg is iterable
+    if hasattr(repository_cfg, '__iter__'):
+        for cfg in repository_cfg:
+            yield _iter_ocm_repositories(component, cfg)
```

Comment:
Should also be `yield from`.

Reply:
why? if repository_cfg is iterable, the for-loop will to a `yield from` already
flattenning is already done via recursion